packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -227,44 +238,55 @@ function arrayToMapTree(
 	};
 }
 
-function mapToMapTree(
-	data: Map<string, InsertableContent>,
-	allowedTypes: ReadonlySet<TreeNodeSchema>,
-): MapTree {
-	const schema = getType(data, allowedTypes);
-	if (schema.kind !== NodeKind.Map) {
-		throw new UsageError(
-			`Provided map input is incompatible with schema "${schema.identifier}".`,
-		);
-	}
+/**
+ * Transforms data under a Map schema.
+ * @param data - The tree data to be transformed. Must be either a TypeScript Map or a Record-like object.
+ * @param schema - The schema associated with the value.
+ * @param allowedTypes - The allowed types specified by the parent.
+ * Used to determine which fallback values may be appropriate.
+ */
+function mapToMapTree(data: InsertableContent, schema: TreeNodeSchema): MapTree {
+	assert(schema.kind === NodeKind.Map, "Expected a Map schema.");
 
-	const childSchema = normalizeAllowedTypes(schema.info as ImplicitAllowedTypes);
+	const allowedChildTypes = normalizeAllowedTypes(schema.info as ImplicitAllowedTypes);
 
-	const fields = new Map<FieldKey, MapTree[]>();
-	for (const [key, value] of data) {
-		assert(!fields.has(brand(key)), 0x84c /* Keys should not be duplicated */);
+	let fields: [string, InsertableContent][] = [];
+	if (data instanceof Map) {
+		fields = [...data.entries()];
+	} else if (typeof data === "object" && data !== null) {
+		fields = Object.entries(data);
+	} else {
+		throw new UsageError(`Input data is incompatible with Map schema: ${data}`);
+	}
+
+	const transformedFields = new Map<FieldKey, MapTree[]>();
+	for (const [key, value] of fields) {
```

Comment:
the current types supported by the API are all Iterable<[string,child]>, so you can do:

```suggestion
	const transformedFields = new Map<FieldKey, MapTree[]>();
	for (const [key, value] of data as Iterable<[string, InsertableContent]>) {
```